src/build.d

```diff
@@ -1867,6 +1896,47 @@ auto run(T)(T args, string workDir = runDir)
     return res.output;
 }
 
+/**
+Wrapper around execute that logs the execution, prints the output,
+and throws an exception for a non-zero exit code.
+
+Params:
+    args = the command and command arguments to execute
+    workDir = the commands working directory
+*/
+void verboseRun(T)(string section, T args, string workDir = runDir)
+{
+    import std.ascii : newline;
+    auto res = pipeProcess(args, Redirect.stdout | Redirect.stderr, null,
+                           Config.none, workDir);
+    int status = wait(res.pid);
+    if (status)
+    {
+        string output;
+        auto stderrRange = res.stderr.byLineCopy;
+        auto stdoutRange = res.stderr.byLineCopy;
+        // only prefix stdout/stderr if both have been used
+        if (!stderrRange.empty && !stdoutRange.empty)
+        {
+            auto stderr = stderrRange.map!(l => "[stderr] " ~ l);
+            auto stdout = stdoutRange.map!(l => "[stdout] " ~ l);
+            output = chain(stderr, stdout).join(newline);
```

Comment:
Not sure about this prefixing. The problem is that the messages logged to stderr and stdout aren't timestamp and thus we can't recreate the exact output. We could also just do:

```
Stderr:
  ...

Stdout:

   ...
```